Ensure CMake picks up the correct OpenMP library (the one belonging to the compiler)

When using GCC `libgomp.so` should be used which will be automatically done with `-fopenmp`.
However the custom FindOpenMP searches for `libomp.so` first which ends up being used if found
e.g. on the system in /lib64

See https://github.com/pytorch/pytorch/pull/177126

Author: Alexander Grund (TU Dresden)

diff --git a/cmake/Modules/FindOpenMP.cmake b/cmake/Modules/FindOpenMP.cmake
--- a/cmake/Modules/FindOpenMP.cmake
+++ b/cmake/Modules/FindOpenMP.cmake
@@ -256,7 +256,7 @@ function(_OPENMP_GET_FLAGS LANG FLAG_MODE OPENMP_FLAG_VAR OPENMP_LIB_NAMES_VAR)
 
     if (NOT OpenMP_libomp_LIBRARY)
       find_library(OpenMP_libomp_LIBRARY
-        NAMES omp gomp iomp5
+        NAMES omp gomp iomp5 NAMES_PER_DIR
         HINTS ${CMAKE_${LANG}_IMPLICIT_LINK_DIRECTORIES}
         DOC "libomp location for OpenMP"
       )
